tensorrt_llm/_torch/speculative/utils.py

```
@@ -97,6 +99,8 @@ def get_spec_resource_manager(model_engine, draft_model_engine=None):
         return NGramPoolManager(spec_config, max_num_requests)
     if spec_dec_mode.is_user_provided():
         return spec_config.resource_manager
+    if spec_dec_mode.is_external_api():
+        return drafter.spec_resource_manager
```

Ruff (0.12.2)

Undefined name `drafter`

(F821)